spinalcordtoolbox/registration/register.py

```diff
@@ -63,6 +67,8 @@ def __init__(self, step=None, type=None, algo='syn', metric='MeanSquares', iter=
         self.type = type
         self.algo = algo
         self.metric = metric
+        self.samplingStrategy = samplingStrategy
+        self.samplingPercentage = samplingPercentage
```

Comment:
Maybe these should be type-checked? At least

```
if samplingStrategy not in ['None','Regular','Random']: raise ValueError(...)
if not (0 <= samplingPercentage <= 1): raise ValueError(...)
```

plus maybe

```
if samplingStrategy == 'None' and samplingPercentage is not None: raise ValueError("samplingPercentage is mutually exclusive with None samplingStrategy")
```

Does ants give reasonable error messages in these cases?

What happens if someone passes a float instead of a string or vice versa?

Reply:
I guess nothing -else- is typechecked in here yet. I don't know.

Reply:
unfortunately i have no more time to dedicate to this PR-- i need to move on-- pls feel free to follow-up in an issue if you think this is relevant